nameresolution/dns/dns_test.go

```diff
@@ -0,0 +1,25 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package dns
+
+import (
+	"testing"
+
+	"github.com/dapr/components-contrib/nameresolution"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestResolve(t *testing.T) {
+	resolver := NewResolver(logger.NewLogger("test"))
+	request := nameresolution.ResolveRequest{ID: "myid", Namespace: "abc", Port: 1234}
```

Comment:
So, does it mean namespace is ignored?

Reply:
Yep. You are right.

Reply:
I should have been clear about my question. But why can't the namespace be part of the dns name? :)

Reply:
Its solution has two reasons
1. When we add a NAMESPACE environment variable Dapr decide that we run in Kubernetes mode and we must set control plane address to proceed
2. This mode (dns) universal in our vision and could be used with another orchestrator (Docker Swarm) or without an orchestrator on different hosts. Question about run Dapr with docker compose on several hosts was two days ago in the general channel in Discord.
Sorry for previous answer.

Reply:
/cc @yaron2  Can you comment on this?

Reply:
@abogdanov37 namespaces are supported for Dapr in self-hosted mode as well. Dapr knows it runs in Kubernetes if the Kubernetes mode flag is specified for the `daprd` process.

Reply:
We had following problem. When we set namespace env variable sidecar with `daprd` can't get configuration from file. Because of that we cant proceed work

Reply:
How can I use namespace in self-hosted mode? I have no ideas(

Reply:
@abogdanov37 - Namespaces in self hosted mode are described here https://docs.dapr.io/operations/components/component-scopes/#namespaces. You set NAMEPSPACE env var.

Reply:
Thank you very much! It changes the World! Based on this I think I can add namespace support. I'll update PR a bit later.

Reply:
But after a bit of thinking, I see a not clear concept about namespaces in a standalone mode Dapr use namespace to separate components but not to finding services. Because of this, I don't want to use namespaces to find services. To find components I can set relevant Kubernetes service name as redis.storage.svc.cluster.local. All concept of my mode based on a single namespace.

Reply:
Dapr uses namespace to find services too - it allows cross namespace service invocation. Namespace is just a logical separation. Can that be helpful in your scenario?